kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/test/TestCaseConfig.kt

```diff
@@ -31,7 +53,9 @@ data class TestCaseConfig(
    val listeners: List<TestListener> = emptyList(),
    val extensions: List<TestCaseExtension> = emptyList(),
    val enabledIf: EnabledIf = { true },
-   val severity: TestCaseSeverityLevel? = null
+   val severity: TestCaseSeverityLevel? = null,
+   val enabledOrReason: Enabled = Enabled.enabled,
```

Comment:
Do we even need the non function version? The only reason we have both for enabled is because it was added before I added the function version.